src/java/com/sarality/app/view/dialog/DialogActionPerformer.java

```diff
@@ -0,0 +1,25 @@
+package com.sarality.app.view.dialog;
+
+import android.content.DialogInterface;
+
+import com.sarality.app.view.action.Action;
+
+/**
+ * Utility class to perform the action.
+ * <p/>
+ * This is the class that gets registered with the dialog and delegates the call to the actual action.
+ *
+ * @author abhideep@ (Abhideep Singh)
+ */
+class DialogActionPerformer implements DialogInterface.OnCancelListener {
```

Comment:
It is meant to handle all actions that require just the dialog. For now we have added support for only Cancel, but will add support for others as well. And hence the name